### PR TITLE
[daum.net] Support for playlists, user channels

### LINKED ISSUE #1952: [daum.net] Add support for playlists (was: Unsupported URL for tvpot.daum.net)
Thanks a lot for your support:)

http://m.tvpot.daum.net/clip/ClipView.tv?clipid=54999425
http://m.tvpot.daum.net/mypot/Top.tv?ownerid=45x1okb1If50&playlistid=3569733

(1)
Jay-Yuui-MacBook-Pro:youtube-dl jcyu5662$ ./youtube-dl -s --verbose http://m.tvpot.daum.net/clip/ClipView.tv?clipid=54999425
[debug] System config: []
[debug] User config: []
[debug] Command-line args: ['-s', '--verbose', 'http://m.tvpot.daum.net/clip/ClipView.tv?clipid=54999425']
[debug] youtube-dl version 2013.12.11.2
[debug] Python version 2.7.5 - Darwin-13.0.0-x86_64-i386-64bit
[debug] Proxy map: {}
WARNING: Falling back on generic information extractor.
[generic] ClipView.tv?clipid=54999425: Downloading webpage
[generic] ClipView.tv?clipid=54999425: Extracting information
ERROR: Unsupported URL: http://m.tvpot.daum.net/clip/ClipView.tv?clipid=54999425; please report this issue on https://yt-dl.org/bug . Be sure to call youtube-dl with the --verbose flag and include its complete output. Make sure you are using the latest version; type  youtube-dl -U  to update.
Traceback (most recent call last):
  File "./youtube-dl/youtube_dl/YoutubeDL.py", line 471, in extract_info
    ie_result = ie.extract(url)
  File "./youtube-dl/youtube_dl/extractor/common.py", line 134, in extract
    return self._real_extract(url)
  File "./youtube-dl/youtube_dl/extractor/generic.py", line 254, in _real_extract
    raise ExtractorError(u'Unsupported URL: %s' % url)
ExtractorError: Unsupported URL: http://m.tvpot.daum.net/clip/ClipView.tv?clipid=54999425; please report this issue on https://yt-dl.org/bug . Be sure to call youtube-dl with the --verbose flag and include its complete output. Make sure you are using the latest version; type  youtube-dl -U  to update.

(2)

Jay-Yuui-MacBook-Pro:youtube-dl jcyu5662$ ./youtube-dl -s --verbose http://m.tvpot.daum.net/mypot/Top.tv?ownerid=45x1okb1If50&playlistid=3569733
[1] 30832
Jay-Yuui-MacBook-Pro:youtube-dl jcyu5662$ [debug] System config: []
[debug] User config: []
[debug] Command-line args: ['-s', '--verbose', 'http://m.tvpot.daum.net/mypot/Top.tv?ownerid=45x1okb1If50']
[debug] youtube-dl version 2013.12.11.2
[debug] Python version 2.7.5 - Darwin-13.0.0-x86_64-i386-64bit
[debug] Proxy map: {}
WARNING: Falling back on generic information extractor.
[generic] Top.tv?ownerid=45x1okb1If50: Downloading webpage
[generic] Top.tv?ownerid=45x1okb1If50: Extracting information
ERROR: Unsupported URL: http://m.tvpot.daum.net/mypot/Top.tv?ownerid=45x1okb1If50; please report this issue on https://yt-dl.org/bug . Be sure to call youtube-dl with the --verbose flag and include its complete output. Make sure you are using the latest version; type  youtube-dl -U  to update.
Traceback (most recent call last):
  File "./youtube-dl/youtube_dl/YoutubeDL.py", line 471, in extract_info
    ie_result = ie.extract(url)
  File "./youtube-dl/youtube_dl/extractor/common.py", line 134, in extract
    return self._real_extract(url)
  File "./youtube-dl/youtube_dl/extractor/generic.py", line 254, in _real_extract
    raise ExtractorError(u'Unsupported URL: %s' % url)
ExtractorError: Unsupported URL: http://m.tvpot.daum.net/mypot/Top.tv?ownerid=45x1okb1If50; please report this issue on https://yt-dl.org/bug . Be sure to call youtube-dl with the --verbose flag and include its complete output. Make sure you are using the latest version; type  youtube-dl -U  to update.

### PR DESCRIPTION
**Pull request by ping**
Fixes #1952

And also fixes error when trying to download http://m.tvpot.daum.net/v/65139429

```
$ youtube-dl -v "http://m.tvpot.daum.net/v/65139429"
[debug] System config: []
[debug] User config: [u'--no-mtime', u'-o', u'/Users/ping/Downloads/%(title)s_%(id)s.%(ext)s']
[debug] Command-line args: [u'-v', u'http://m.tvpot.daum.net/v/65139429']
[debug] Encodings: locale UTF-8, fs utf-8, out UTF-8, pref UTF-8
[debug] youtube-dl version 2016.01.15
[debug] Python version 2.7.5 - Darwin-13.4.0-x86_64-i386-64bit
[debug] exe versions: ffmpeg 2.6.2, ffprobe 2.6.2, rtmpdump 2.4
[debug] Proxy map: {}
[daum.net] 65139429: Downloading video info
[daum.net] 65139429: Downloading video formats info
ERROR: An extractor error has occurred. (caused by KeyError(u'output_list',)); please report this issue on https://yt-dl.org/bug . Make sure you are using the latest version; type  youtube-dl -U  to update. Be sure to call youtube-dl with the --verbose flag and include its complete output.
Traceback (most recent call last):
  File "/usr/local/bin/youtube-dl/youtube_dl/extractor/common.py", line 312, in extract
    return self._real_extract(url)
  File "/usr/local/bin/youtube-dl/youtube_dl/extractor/daum.py", line 46, in _real_extract
    for format_el in movie_data['output_list']['output_list']:
KeyError: u'output_list'
Traceback (most recent call last):
  File "/usr/local/bin/youtube-dl/youtube_dl/YoutubeDL.py", line 665, in extract_info
    ie_result = ie.extract(url)
  File "/usr/local/bin/youtube-dl/youtube_dl/extractor/common.py", line 318, in extract
    raise ExtractorError('An extractor error has occurred.', cause=e)
ExtractorError: An extractor error has occurred. (caused by KeyError(u'output_list',)); please report this issue on https://yt-dl.org/bug . Make sure you are using the latest version; type  youtube-dl -U  to update. Be sure to call youtube-dl with the --verbose flag and include its complete output.
```

### CODE REVIEW

File: youtube_dl/extractor/daum.py [hunk 1]
```diff
@@ -110,3 +130,115 @@ def _real_extract(self, url):
             'view_count': int_or_none(clip_info.get('play_count')),
             'ie_key': 'Daum',
         }
+
+
+class DaumListIE(InfoExtractor):
+    def _get_entries(self, list_id, list_id_type):
+        name = None
+        entries = []
+        for pagenum in itertools.count(start=1):
```
Comment (yan12125) [inline youtube_dl/extractor/daum.py]: ```itertools.count``` does not accept keyword arguments until Python 2.7.

File: youtube_dl/extractor/daum.py [hunk 2]
```diff
@@ -110,3 +130,115 @@ def _real_extract(self, url):
...
+        },
+        'params': {
+            'noplaylist': True,
+            'skip_download': True,
+        }
+    }]
+
+    def _real_extract(self, url):
+        if DaumClipIE.suitable(url) and self._downloader.params.get('noplaylist'):
+            return self.url_result(url, 'DaumClip')
```
Comment (yan12125) [inline youtube_dl/extractor/daum.py]: It's better to have some tips for users that ```--no-playlist``` is applicable here, like [what ```YoutubeIE``` does](https://github.com/rg3/youtube-dl/blob/2975fe1a7be39b66d14fc7ef68ea7db948f0e40f/youtube_dl/extractor/youtube.py#L1764).

File: youtube_dl/extractor/daum.py [hunk 3]
```diff
@@ -110,3 +130,115 @@ def _real_extract(self, url):
...
+            'id': 'o2scDLIVbHc0',
+            'title': '마이 리틀 텔레비전',
+        },
+        'playlist_mincount': 213
+    }, {
+        'url': 'http://tvpot.daum.net/mypot/View.do?ownerid=o2scDLIVbHc0&clipid=73801156',
+        'info_dict': {
+            'id': '73801156',
+            'ext': 'mp4',
+            'title': '[미공개] 김구라, 오만석이 부릅니다 &#39;오케피&#39; - 마이 리틀 텔레비전 20160116',
```
Comment (yan12125) [inline youtube_dl/extractor/daum.py]: &#39; should be unescaped to single quotes to match the title shown on the webpage.

**Comment by yan12125** [comment]
Could you separate these change into two commits - fixes error when trying to download http://m.tvpot.daum.net/v/65139429 and support for playlists, user channels?

File: youtube_dl/extractor/daum.py (hunk 2 above)
Comment (ping) [inline youtube_dl/extractor/daum.py]: Done.

**Comment by yan12125** [comment]
Could you squash all commits into a single one and re-push?

**Comment by ping** [comment]
Made a bunch of mistakes with this. Will redo in a new PR.

**Pull request closed without merging**